src/ssids/cpu/kernels/block_ldlt.hxx

```diff
@@ -298,6 +300,9 @@ void block_ldlt(int from, int *perm, T *a, int lda, T *d, T *ldwork,
       int t, m; // row and col location of maximum entry
       find_maxloc<T,BLOCK_SIZE>(p, a, lda, bestv, t, m);
 
+      // Handle case where find_maxloc failed (e.g., due to NaNs in the input)
+      if (t >= BLOCK_SIZE || m >= BLOCK_SIZE) throw SingularError(p);
```

Comment:
I'm not quite sure about this check. It unconditionally gives up with `SingularError`, ignoring the `action` setting. Also, if `find_maxloc` failed, the change above will cause `bestv` to be 0.0, and so `fabs(bestv) < small` will be true, so we should go into the handling of the singular case anyways, right?

Reply:
But in my tests, the "broken!" error is what I hit without this added check.